src/toil/batchSystems/kubernetes.py

```diff
@@ -497,47 +489,19 @@ def _ourJobObjects(self, onlySucceeded=False, limit=None):
         # https://github.com/kubernetes-client/python/issues/953. For now we
         # will just throw an error if we don't get to the end of the list in
         # time.
-        token = None
-        
-        # Do our own limiting since we need to apply a filter that the server
-        # can't.
-        seen = 0
-        
-        # TODO: We ought to label our jobs by owning Toil workflow so we can
-        # look them up instead of filtering down later.
         
-        while True:
-            # We can't just pass e.g. a None continue token when there isn't
-            # one, because the Kubernetes module reads its kwargs dict and
-            # cares about presence/absence. So we build a dict to send.
-            kwargs = {}
-            if onlySucceeded:
-                # Check only successful jobs.
-                # Note that for selectors it is "successful" while for the
-                # actual object field it is "succeeded".
-                kwargs['field_selector'] = 'status.successful==1'
-            if token is not None:
-                kwargs['_continue'] = token
-            
-            results = self._try_kubernetes(self._api('batch').list_namespaced_job, self.namespace, **kwargs)
-            
-            for job in results.items:
-                if self._isJobOurs(job):
-                    # This job belongs to us
-                    yield job
-                    
-                    # Don't go over the limit
-                    seen += 1
-                    if limit is not None and seen >= limit:
-                        return
-                    
-            # Remember the continuation token, if any
-            token = getattr(results.metadata, 'continue', None)
```

Comment:
With the label, you can rip out the `if self._isJobOurs(job):` logic, because the returned jobs will always be ours. But I don't think you can safely get rid of all this token and `_continue` logic. When you call `list_namespaced_job`, even with a `label_selector`, you don't always get all the jobs in the response, because there could be too many of them to send all at once. Instead, you get the first page of jobs (like Google search results), and you need to make another call, providing the "continuation token" from the first page, in order to get the second page, and so on until the server runs out of pages to send you.

If you want to be able to retrieve jobs and also pods, maybe you can factor out the get-all-the-pages logic in such a way that it can wrap calls to either `list_namespaced_job` or `list_namespaced_pod`, but you need to be able to get all the pages. We could be in a situation where there's a pod that's stuck on the very last page and we need to clean it up before anything else can make progress.

The reason it was an iterator before was so you could work on the pages as they arrive. I guess you could collect all the pages into a single list if you really want to, but it might be a huge list with thousands of items, and using an iterator could be more efficient. The downside of using an iterator is that the continuation tokens eventually expire, so if you spend too much time working on one page, you won't be able to get the next page. So it might indeed be safer to compose the whole list first.